### DISCUSSION TITLE
Projected Texture Improvements

### DISCUSSION THREAD

**Discussion by koerismo** [StrataSource/Portal-2-Community-Edition#500] [Ideas]
**Here are some features that would make projected textures a lot nicer to work with than they are currently:**

## Projected Materials
Allow projected textures to use materials alongside textures. To reference materials instead of textures, users could omit the `.vtf` extension. Material support would be limited to a `ProjectedGeneric` type, which would prevent problems concerning the parameters that exist in the commonly used shaders. (e.g. `$bumpmap`, `$phong`) This would retain the functionality of features such as material proxies, though.

## Build clamping/cutoff into entity keyvalues
Add keyvalues for horizontal and vertical clamping to allow for the following modes:
- Cut: Don't project anything past the texture bounds.
- Clamp: Constrain the texture coordinates to the texture bounds and project to the full extent of the entity's fov.
- Repeat: Repeat the texture forever and project to the full extent of the entity's fov.

## Blend Modes
Add a keyvalue (similar to env_sprite) that controls the blending of the projected texture. This would default to `Additive`.

**Comment by StefanH-AT** [comment]
A projected texture, by definition, projects a texture and not a material. This suggestion will not be worked on

**Reply by koerismo** [reply]
Well, would it be possible to still implement the other suggestions? (Things such as projection blend mode, constant/linear/quadratic values, and having the texture clamp flags be overrideable by entity keyvalues)

The only reason I suggested moving to vmt would be to establish some continuity, as right now it feels really rough around the edges.

**Reply by vrad-exe** [reply]
Projected VMTs would be really useful, please don't refuse to implement something just because it conflicts with a name.
